dice_ml/explainer_interfaces/dice_genetic.py

```diff
@@ -461,47 +371,64 @@ def find_counterfactuals(self, query_instance, desired_range, desired_class, fea
                 stop_cnt += 1
             else:
                 stop_cnt = 0
             if stop_cnt >= 5:
                 break
             previous_best_loss = current_best_loss
-            population_fitness = []
-            current_best_loss = np.inf
-            current_best_cf = []
-            for k in range(self.population_size):
-                loss = self.compute_loss(population[k], desired_range, desired_class)
-                population_fitness.append((k, loss))
-
-                if loss < current_best_loss:
-                    current_best_loss = loss
-                    current_best_cf = population[k]
-
-            cfs_preds = [self.predict_fn(cfs) for cfs in current_best_cf]
-            # 10% of the next generation is fittest members of current generation
-            population_fitness = sorted(population_fitness, key=lambda x: x[1])
-            s = int((10 * self.population_size) / 100)
-            new_generation = [population[tup[0]] for tup in population_fitness[:s]]
-
-            # 90% of the next generation obtained from top 50% of fittest members of current generation
-            s = int((90 * self.population_size) / 100)
+
+            new_array = [tuple(row) for row in population]
+            population = np.unique(new_array, axis=0)
+
+            population_fitness = self.compute_loss(population, desired_range, desired_class)
+            population_fitness = population_fitness[population_fitness[:, 1].argsort()]
+
+            current_best_loss = population_fitness[0][1]
+            to_pred = np.array([population[int(tup[0])] for tup in population_fitness[:self.total_CFs]])
+            cfs_preds = self.predict_fn(to_pred)
+
+            # self.total_CFS of the next generation obtained from the fittest members of current generation
+            s = self.total_CFs
+            new_generation_1 = np.array([population[int(tup[0])] for tup in population_fitness[:s]])
+
+            # rest of the next generation obtained from top 50% of fittest members of current generation
+            s = self.population_size - s
+            new_generation_2 = np.zeros((s, self.data_interface.number_of_features))
             for _ in range(s):
-                parent1 = random.choice(population[:int(50 * self.population_size / 100)])
-                parent2 = random.choice(population[:int(50 * self.population_size / 100)])
+                parent1 = random.choice(population[:int(len(population) / 2)])
+                parent2 = random.choice(population[:int(len(population) / 2)])
                 child = self.mate(parent1, parent2, features_to_vary, query_instance)
-                new_generation.append(child)
+                new_generation_2[_] = child
```

Comment:
usually `_` is only used if you don't want to use the variable.  here use something like i or j